### DISCUSSION TITLE
Non-Mathesar Data Types

### DISCUSSION THREAD

**Discussion by ghislaineguerin** [mathesar-foundation/mathesar#551] [Maintainer discussions]
The specs related to issue #537 are ready for review. Please take a look at the attached document and follow the [review guidelines](https://wiki.mathesar.org/en/design/process/review-guidelines).

## Document Link
https://wiki.mathesar.org/en/design/specs/non-mathesar-data-types

@mathemancer @kgodey @pavish, your feedback is required by Thursday 19th. Please plan your review so that at least one round of changes can be made and reviewed if needed.

**Comment by kgodey** [comment]
@ghislaineguerin This doesn't address some of the requirements from #537. 

Specifically, it does not make a distinction between `other` types (which are default Postgres types that we know about but don't handle yet) and `custom` types (which are completely unknown types). It also doesn't convey to the user **_why_** they can't make any changes to these types.

**Reply by ghislaineguerin** [reply]
Maybe it's not clear in the prototype. I've added different icons for each and a tooltip for explaining why they can't access additional options. I might need to make this more evident. My impression was that everything else would be similar to how we handle string values. I will add additional content so that the difference is more obvious to users.
![image](https://user-images.githubusercontent.com/845767/129210665-b4971835-f462-4584-8c5a-c71222020a9e.png)

**Reply by ghislaineguerin** [reply]
@kgodey for the unknown types, do we not know their type name?

**Reply by kgodey** [reply]
We do know the type name.

**Reply by kgodey** [reply]
@ghislaineguerin Sorry, I missed the "Other" type and the tooltip.

Some more feedback:
- I think the icons for custom/other should be bigger or clearer, they're hard to make out.
- The messages explaining the data type should be more prominent.
  - I don't think "aren't supported by Mathesar" makes sense - they are supported, they can see and edit them. They just can't change the data type.
  - I don't think "don't have any special functionality" will be a useful message for the user – what is special functionality?
- The messages for each data type should be different. 
- It would be good to show more of the values for columns with longer text like the JSON column.

**Reply by ghislaineguerin** [reply]
@kgodey When you say 'show more of the values' do you mean we should allow for multi-line text within cells? Here's an example. If so, what would the general rule be to allow cells to expand? Should we have display options for the table where users can set the desired cell size?
![image](https://user-images.githubusercontent.com/845767/130630165-2ed0b8b3-b87a-4a87-b92e-7cedcf7a215f.png)

**Reply by kgodey** [reply]
@ghislaineguerin I'm not sure if we need to support users setting cell size for the MVP, I think it makes sense to dynamically adjust them based on data size. We could support different table display settings in the future.

**Comment by ghislaineguerin** [comment]
@kgodey for the explanation messages, please let me know if this is correct:
Custom types cannot be changed as they are defined outside of Mathesar.
Other types cannot be changed as they aren't mapped to a Mathesar built-in type.

**Reply by kgodey** [reply]
@ghislaineguerin Yes, that's correct.
